### PR TITLE
Add 6443 port description

### PR DESCRIPTION
**Pull request by jichenjc**
**What this PR does / why we need it**:

6443 port actually is hard coded...  so we should warn user they can't update it

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

1. Please confirm that if this PR changes any image versions, then that's the sole change this PR makes.

**Release note**:

```release-note

```

### CODE REVIEW

File: docs/configuration.md [hunk 1]
```diff
@@ -19,6 +19,7 @@
   - [Boot From Volume](#boot-from-volume)
   - [Timeout settings](#timeout-settings)
   - [Custom pod network CIDR](#custom-pod-network-cidr)
+  - [configuration of KubeadmControlPlane](#configuration-of-kubeadmcontrolplane)
```
Comment (hidekazuna) [inline docs/configuration.md:22]: Please capitalize the first letter.

File: docs/configuration.md [hunk 2]
```diff
@@ -229,3 +230,10 @@ If creating servers in your OpenStack takes a long time, you can increase the ti
 ## Custom pod network CIDR
 
 If `192.168.0.0/16` is already in use within your network, you must select a different pod network CIDR. You have to replace the CIDR `192.168.0.0/16` with your own in the generated file.
+
+## configuration of KubeadmControlPlane
+`KubeadmControlPlane` is defined [here](https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_types.go#L146) and you need to define it to create a kubeadm control plane.
+
+`cluster-api-openstack-provider` defined templates for you at [with-lb](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/templates/cluster-template.yaml) and [without-lb](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/templates/cluster-template-without-lb.yaml), you can go ahead and do additional configuration.
+
+Please note the port `6443` must be kept at `controlPlaneEndpoint: "${OPENSTACK_CONTROLPLANE_IP}:6443"` because the security group is hard coded at [here](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/pkg/cloud/services/networking/securitygroups.go#L136).
```
Comment (hidekazuna) [inline docs/configuration.md:239]: If you want to use the other port, raise an issue and fix, please.

File: docs/configuration.md [hunk 3]
```diff
@@ -229,3 +230,10 @@ If creating servers in your OpenStack takes a long time, you can increase the ti
 ## Custom pod network CIDR
 
 If `192.168.0.0/16` is already in use within your network, you must select a different pod network CIDR. You have to replace the CIDR `192.168.0.0/16` with your own in the generated file.
+
+## configuration of KubeadmControlPlane
+`KubeadmControlPlane` is defined [here](https://github.com/kubernetes-sigs/cluster-api/blob/master/controlplane/kubeadm/api/v1alpha3/kubeadm_control_plane_types.go#L146) and you need to define it to create a kubeadm control plane.
+
+`cluster-api-openstack-provider` defined templates for you at [with-lb](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/templates/cluster-template.yaml) and [without-lb](https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/templates/cluster-template-without-lb.yaml), you can go ahead and do additional configuration.
```
Comment (hidekazuna) [inline docs/configuration.md:237]: What is "additional configuration"?

File: docs/configuration.md (hunk 1 above)
Comment (jichenjc) [reply docs/configuration.md:22]: done

File: docs/configuration.md (hunk 2 above)
Comment (jichenjc) [reply docs/configuration.md:239]: I understand it's by default ,so just provide some info 
I am thinking how to fix it , we might need additional configuration

File: docs/configuration.md (hunk 3 above)
Comment (jichenjc) [reply docs/configuration.md:237]: such as cloud specific settings ..I will make it more clear

File: docs/configuration.md (hunk 1 above)
Comment (hidekazuna) [reply docs/configuration.md]: Thanks, confirmed.

File: docs/configuration.md (hunk 2 above)
Comment (hidekazuna) [reply docs/configuration.md:239]: We can raise an issue and add `good first issue` tag for new contributors. This is not difficult.

**Comment by hidekazuna** [comment]
Now listening port is configurable by openStackCluster.Spec.APIServerPort.

https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/master/controllers/openstackcluster_controller.go#L411

**Pull request closed without merging**